packages/client-react/src/SwrRequest.ts

```diff
@@ -0,0 +1,16 @@
+import { PureModel } from '@datx/core';
+import { INetwork, Request, Response, IAsync, IGeneralize } from '@datx/network';
+
+export class SwrRequest<
+  TNetwork extends INetwork,
+  TModel extends typeof PureModel,
+  TResponse extends InstanceType<TModel> | Array<InstanceType<TModel>>,
+  IA extends IAsync<InstanceType<TModel>> = IAsync<InstanceType<TModel>>,
+> extends Request<TNetwork, TModel, TResponse, IA> {
+  public swr(): { key: string; fetcher: () => IGeneralize<Response<InstanceType<TModel>>, IA> } {
```

Comment:
I don't think this is the best solution.
Some things were changed in SWR form the last time we talked about this.
SWR now support objects as a key https://swr.vercel.app/docs/arguments#passing-objects
```
const { data: orders } = useSWR({ url: '/api/orders', args: user }, fetcher)
```
So the best option here is to return `Expression`, for example https://github.com/infinum/datx/blob/373-ssr-support/packages/swr/src/interfaces/QueryExpression.ts#L14

flow goes like this:
1. `useQuery({ op: 'getOne', model: 'planet', id: 1 })` or [conditional fetching](https://swr.vercel.app/docs/conditional-fetching) `useQuery(() => id ? { op: 'getOne', model: 'planet', id } : null)`
2. It will go through middleware which will pass the expression to fetcher and extend SWR with networkConfig feature https://github.com/infinum/datx/blob/373-ssr-support/packages/swr/src/middleware.tsx#L6
3. Fetcher will just compile the expression and call the correct method on the client https://github.com/infinum/datx/blob/373-ssr-support/packages/swr/src/createFetcher.ts#L31

This way we decoupled networking layer and query builder, we can then build adapters that could interpret expression differently.

For JSON:API, QueryBulder could only build the a URL.

Later on we can think about one query accumulated expression that could be an array of operations, something like  atomic operations https://jsonapi.org/ext/atomic/#auto-id-processing-multiple-operations.

Or the thing with chaining:
```
[{ 
  op: 'getOne', 
  model: 'post', 
  id: 1,
  next: (data) => ({ op: 'getMany', model: 'example', filter: { post_id: data.map(getId) }),
},
{
  op: 'getMany',
  model: 'comments',
  filter: { postId : 1 }
}]
```
With next function we can support chaining, but it's clear that this will cause a waterfall effect. 
I want to avoid waterfalls, at least, user should self consciously implement waterfalls, it should not be hidden behind some syntax sugar.

Reply:
Some other API ideas:

Based on this lib https://github.com/pmndrs/suspend-react
```
const data = datx((q) => q.from(ModelB).id('1'));
```

For subscribe:
```
datx(
  (q) => q.from(ModelA).id('1')
).subscribe((resp) => {
    // @ts-expect-error
    console.log(resp.data?.b);
  });
```

Reply:
For the first comment:
This is all doable with the current implementation - the `Request` class decides on the network implementation and it doesn't need to use the fetch method - the requirement is that it has a request method which returns the API response in asynchronous matter.

For the second comment:
This is basically the same API, but with more flexibility:
* The app can have multiple clients
* Promise or Observable - not sure how your example would know which one to use? If you have a separate (global?) config for this, the correct types would be impossible to implement.
* Each client can use a different API standard (you could have a JSON:API client, a HAL client and a custom REST client with identical interfaces to the client user)